core/src/test/scala/org/apache/spark/sql/delta/LogStoreSuite.scala

```diff
@@ -220,9 +218,8 @@ class AzureLogStoreSuite extends LogStoreSuiteBase {
   protected def shouldUseRenameToWriteCheckpoint: Boolean = true
 }
 
-class HDFSLogStoreSuite extends LogStoreSuiteBase {
+trait HDFSLogStoreSuiteBase extends LogStoreSuiteBase {
```

Comment:
QQ: why do we need a base class?

Reply:
it just makes more sense that we have some core functionality X. Then we want to test that with the internal (scala) hdfs impl and the external (java) hdsfs impl.

So `HDFSLogStoreSuite extends HDFSLogStoreSuiteBase` and `PublicHDFSLogStoreSuite extends HDFSLogStoreSuiteBase` is the cleanest way to do that.

If we had `PublicHDFSLogStoreSuite extends HDFSLogStoreSuite` (no `Base)` then that would look weird IMO.